anchore_engine/services/policy_engine/engine/feeds/feeds.py

```diff
@@ -906,26 +907,84 @@ def _switch_active_grypedb(
         )
         db.add(self.grypedb_meta)
 
-    def _process_group_file_records(
+    @staticmethod
+    def _run_grypedb_sync_task(
+        db: Session, checksum: str, grype_db_data: bytes
+    ) -> None:
+        """
+        Write the Grype DB to a tar.gz in a temporary directory and pass to GrypeDBSyncManager.
+        The GrypeDBSyncManager updates the working copy of GrypeDB on this instance of policy engine.
+
+        :param checksum: grype DB file checksum
+        :type checksum: str
+        :param grype_db_data: raw tar.gz file data
+        :type grype_db_data: bytes
+        """
+        with GrypeDBStorage() as grypedb_file:
+            with grypedb_file.create_file(checksum) as f:
+                f.write(grype_db_data)
+            GrypeDBSyncManager.run_grypedb_sync(db, grypedb_file.path)
+
+    def _set_group_counts(self, db, source_counts: List[RecordSource]) -> None:
+        """
+        Uses grype wrapper to query the source counts in the grypedb
+        casts these source counts to FeedGroupMetadata and saves it to GrypeDBFeedMetadata groups column
+        certain timestamps like the last_sync are not set in this process the first time around because it has not yet
+        been set on the GrypeDBFeedMetadata object
+        """
+        groups = []
+
+        for source in source_counts:
```

Comment:
Perhaps it would be nice to log this information for debug purposes so that one can see the sort of group info that we are populating during the sync?